### PR TITLE
update to release 1.0.0

### PR DESCRIPTION
**Pull request by albertborsos**
- validate sentry DSN even if the component is disabled
- DSN is required even if the component is disabled
- default environment tag is production
- remove deprecated methods and properties
- separate init method's content to dividual methods

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+2016-11-23 (aborsos) - 1.0.0.
+ - validate sentry DSN even if the component is disabled
+ - DSN is required even if the component is disabled
+ - default environment tag is `production`
+ - remove deprecated methods and properties
+ - separate init method's content to dividual methods
```
Comment (nkovacs) [inline CHANGELOG.md]: individual

**Comment by erickskrauch** [comment]
Few suggestions:

* Add possibility to define some callback, that will be called with `$context` argument. All returned data will be merged into `$data` variable.

* `SentryComponent` should implement some `SentryComponentInerface`with public methods `capture`, `captureException`, `captureMessage` and `isEnabled`. Inside `SentryTarget` we must call `Instance::ensure` on that interface.

* You running tests on php 5.6+ version. My suggestion is to use `::class` (php 5.5+) instead of [string class definition](https://github.com/hellowearemito/yii2-sentry/pull/4/files#diff-1810474ed820edd09e1d8964ae0826e3R104).

* Cleanup tests folder (maybe you not need codeception and can just use phpunit).

Also little proposal: rename classes into clear `Component` and `Target`. They are in namespace `mito\sentry`, so `Sentry`-prefix is not required.

------

I can help you with this by submitting PR, if you have little time. This component just is needed inside my app.

**Comment by nkovacs** [comment]
> Add possibility to define some callback, that will be called with `$context` argument. All returned data will be merged into `$data` variable.

I'm not sure how you would use this. The goal with this extension was to create a drop-in log target, and those don't have a callback like this. They do have a `getContextMessage` method (which just logs globals defined in `logVars`) that we had to [disable](https://github.com/hellowearemito/yii2-sentry/blob/master/src/SentryTarget.php#L40) because it would generate a separate message. We could bring back `logVars` support, but send it as one message.

> `SentryComponent` should implement some `SentryComponentInerface` with public methods `capture`, `captureException`, `captureMessage` and `isEnabled`. Inside `SentryTarget` we must call `Instance::ensure` on that interface.

We don't use `captureMessage` in `SentryTarget`, so I don't think we should include it in the interface.

> Also little proposal: rename classes into clear `Component` and `Target`. They are in namespace `mito\sentry`, so `Sentry`-prefix is not required.

:+1:

**Comment by erickskrauch** [comment]
> I'm not sure how you would use this.

Send current application version or build number + user identity, if it presented.

> We don't use `captureMessage` in `SentryTarget`.

But this method presented in `SentryComponent`. Interface in this case needed to mock component.

**Comment by albertborsos** [comment]
You can configure tags, to set version, build, user identity, etc..
Like here:
https://github.com/hellowearemito/yii2-sentry/blob/master/src/SentryComponent.php#L131

**Comment by nkovacs** [comment]
> Send current application version or build number + user identity, if it presented.

This is a more generic problem with Yii's logging (I've run into a similar problem with `EmailTarget` for example), so I've opened a proposal for Yii 2.1: https://github.com/yiisoft/yii2/issues/13103
In the meantime, we could add a callback like you suggested, but that should be a separate issue. Feel free to submit a PR, we'd be grateful.

> But this method presented in `SentryComponent`. Interface in this case needed to mock component.

Mockery allows us to mock the class, we don't need the interface for that.
The interface is more useful if you want to use a custom `SentryComponent` with our `SentryTarget`, and don't want to extend from our `SentryComponent`. In that case we don't need the `captureMessage` method in the interface, because `SentryTarget` does not use it (and I don't think this would be a very common use case, so I'm not entirely convinced we need the interface at all, but it is cleaner than coupling `SentryTarget` to `SentryComponent`).

**Comment by erickskrauch** [comment]
> so I've opened a proposal for Yii 2.1: yiisoft/yii2#13103

👍 

> Feel free to submit a PR, we'd be grateful.

Accepted.

> so I'm not entirely convinced we need the interface at all

This is just suggestion, but interface will not spoil code :)

**Comment by erickskrauch** [comment]
Hello again.
Bits strange behavior of [init step](https://github.com/hellowearemito/yii2-sentry/pull/4/files#diff-c694a29ec41709ca1a5f5e8a29f6467eR63). If component not enabled (disabled for tests), so why I need to validate dsn?

**Pull request merged**